plugins/out_stackdriver/stackdriver.c

```diff
@@ -1229,12 +1263,16 @@ static int stackdriver_format(struct flb_config *config,
         msgpack_pack_str_body(&mp_pck, "timestamp", 9);
 
         /* Format the time */
-        gmtime_r(&tms.tm.tv_sec, &tm);
-        s = strftime(time_formatted, sizeof(time_formatted) - 1,
-                     FLB_STD_TIME_FMT, &tm);
-        len = snprintf(time_formatted + s, sizeof(time_formatted) - 1 - s,
-                       ".%09" PRIu64 "Z", (uint64_t) tms.tm.tv_nsec);
-        s += len;
+        if (tms_status != FORMAT_TIME) {
```

Comment:
Should be explicit checking the cases you would like to format - I think this condition applies to all the tms_status, including the case for `invalid` and `non_exist`.

What are the cases you would like to format? I think they are `FORMAT_TIMESTAMP` and `FORMAT_TIMESTAMPSECONDS`?

Reply:
If the time format is not `time`, we will set the `tms`. If the format is `time`, we can directly set the `flb_sds_t format_time`.

But I haven't supported format `time` yet.